### ISSUE TITLE
Pyenv can't find command which got installed with system's pip

### ISSUE THREAD

**Issue by ohcibi** [pyenv/pyenv#1204]
### Description
- [x] Platform information (e.g. Ubuntu Linux 16.04): Ubuntu 16.04
- [x] pyenv version: 1.2.6
- [x] Python version: system: 2.7.12, 3.5.0, installed with pyenv: 3.6.0
- [x] C Compiler information (e.g. `gcc (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609`): 
- [x] Please attach verbose build log as gist
  * specific build is not (should not) be involved

I have installed `pylint` (using this as an example, **issue is not specific to the pylint package**) with my system python's pip. I'm not 100% sure if my interpretation of the way pyenv works is correct, but here is what I observe:

* `pip3 --version` and `python3 --version` show the systems versions
    * assume that `pip3 install foo` would install the `foo` package into the system's python's `site-packages`, as well as that `pip3 list` would list me all the system's python's `site-packages` and none of any python that got installed with pyenv)
* `pip3 list` lists the `pylint` package

Problem:

    $ pylint
    pyenv: pylint: command not found

    The `pylint` command exists in these Python versions:
      3.6.0

So I found out that a part of this issue is that the shim's `pylint` command is shadowing the `pylint` command that is installed from pip itself (at `$HOME/.local/bin`) which would be no problem if pyenv would actually recognize the `pylint` that was installed from system python and choose that, hence `system` ist currently on of the active versions.

#### Tl;dr of this issue
Pyenv shims should properly detect if packages are installed in the system python's `site-packages`.

**Comment by ohcibi** [comment]
> From what I've seen nothing should ever be installed in a "system"

I'm not sure if this is a misunderstanding but when I say "system" I mean the "system" that pyenv refers to in e.g. `pyenv versions`. If I set my global (or local or shell) interpreter to `system` I of course expect the package to be installed in the `site-packages` of the system's python (I'm installing packages with `--user` mostly, so the packages are not actually in a "system"s folder outside my $HOME directory, if you mean that)

> If yours say that no specific version has been configured then you need to set one... otherwise the shell will search your $PATH and fine the system python which is not what you want.

Sorry but this is exactly what I want. Let me try to explain it differently:

    $ pyenv versions
    * system
    3.6.0
    3.6.6

So I have installed the versions 3.6.0 and 3.6.6. alongside my systems python

    $ pyenv shell 3.6.0
    $ pip list
    ...
    pylint    2.1.1 
    ...
    $ pylint --version
    pylint 2.1.1
    ...
    Python 3.6.0

So when setting the current python to `3.6.0` I expect packages installed with the pip that comes with `3.6.0` to be listed and useable which works the way you described for that pyenv-installed `3.6.0` version.

However as `system` is also considered a "version", I expect the following to work in the same way and not with the error message that comes at the end:

    $ pyenv shell system
    $ pip list
    ...
    pylint    1.9.2
    ...
    $ pylint --version
    pyenv: pylint: command not found

    The `pylint` command exists in these Python versions:
      3.6.0

 So is this a misunderstanding of the two of us or are you basically telling that one cannot use the same python packages in different versions while pyenv chooses the "right one" according to the currently activated interpreter?

P.S.: After I finished writing this, I noticed that `pyenv shell system` in fact makes `pyenv shell` not say "no local version configured...." but simply tell "system". So obviously it was not the missing explicit `pyenv shell system` as I can reproduce my issue with that as well.

**Comment by richard-scott** [comment]
Exactly that, if you need a pip package in a different version of Python you need to change Python version with pyenv local/shell/global etc and then install the package again.

**Comment by richard-scott** [comment]
Also, that "system" version of Python only seems to show up on a Mac, I don't get to see that in Linux.

**Comment by ohcibi** [comment]
> Exactly that, if you need a pip package in a different version of Python you need to change Python version with pyenv local/shell/global etc and then install the package again.

Well that explains my confusion but I'd consider that a bad implementation. First of let me rephrase the way you said what I actually want: 

Instead that I

> need a pip package in a different version of python

I actually want to

> make pyenv find and use a pip package that is installed for the "system" python version if that "system" version is currently activated (e.g. by calling `pyenv shell system`)

Again: if you don't have a `system` python, your pyenv setup seems to be broken and you might fix that first before trying to reproduce my issue (I'm also on linux the whole time, see last comment).
If my pyenv wasn't aware of my "system" python installation I wouldn't expect it to pick up any packages with its shims for that and instead behave exactly the way you described. My whole expectation for that behaviour comes from the fact that pyenv in fact does recognizes my system's pyton and I therefore expect it to pick up it's packages as well.

P.S. please also note that the README.md of this repository also refers to "system python" and even shows an example command to activate multiple python versions where system is one of them:

> `pyenv global system 3.3.6 3.2 2.5`

This is basically the command that I ran when I setup pyenv initially. I just use `3.6.0` as a second argument so I have:

`python` => system's `python2`
`python3` => system's `python3` (which is 3.5 on ubuntu 16.04)
`python3.6` => 3.6.0's `python3`

as a default setup.

**Comment by richard-scott** [comment]
@ohcibi - The lack of a "system" version listed is a bug with pyenv as I get a "system" version on Ubuntu 16, but not on newer Ubuntu versions (e.g. 17 or 18).

**Comment by ohcibi** [comment]
@richard-scott yes I'm also on Ubuntu 16.04. and haven't tested it yet on a newer version.

But in fact thats another bug. This bug here is merely about "system" packages that are not found by the shims pyenv adds to `$PATH`. Is the bug clear now? Should I update the initial issue description somehow to make it more clear and avoid the confusion you had in the beginning?

**Comment by eelkevdbos** [comment]
@ohcibi I ran into the same issue (but on OSX). I found that an old package was already 'satisfying' the dependency and thus not installing a newer version. The old package location wasn't added to the PATH.

I had to remove my old packages (installed with pip before using pyenv). The location is hinted when you try to install it via pip (`pip install pylint` in your case). After removal of this package location you can run your install command one last time to install it to the correct path (included in your system's PATH).

**Comment by ohcibi** [comment]
Just to ensure that I understand correctly: you are saying that one needs to install system packages after installing pyenv in order for pyenv to pick it up correctly? Need to try that home when I’m at the office.

**Comment by eelkevdbos** [comment]
> Just to ensure that I understand correctly: you are saying that one needs to install system packages after installing pyenv in order for pyenv to pick it up correctly? Need to try that home when I’m at the office.

Yes, but not only that. If you've installed packages in a pre-pyenv era, you should make sure that these site-package directories are removed. Pip hints at these directories when you try to install a package by mentioning something along the lines of "Requirement already satisfied: pylint in {{not_your_pyenv_shim_dir}}"

**Comment by native-api** [comment]
> I have installed `pylint` (using this as an example, **issue is not specific to the pylint package**) with my system python's pip. I'm not 100% sure if my interpretation of the way pyenv works is correct, but here is what I observe:
> 
>     * `pip3 --version` and `python3 --version` show the systems versions
>       
>       * assume that `pip3 install foo` would install the `foo` package into the system's python's `site-packages`, as well as that `pip3 list` would list me all the system's python's `site-packages` and none of any python that got installed with pyenv)
> 
>     * `pip3 list` lists the `pylint` package

Ubuntu `pip3` installs packages under `~/.local` -- in particular, executables under `~/.local/bin`.
This directory is added to `PATH` by the stock `~/.profile` -- but only if it exists.
So you need to fully log out and log back in to see the newly-installed `pylint` if this directory didn't exist before that.
